angular-with-directus/how to integrate directus with Angular.md

```diff
@@ -0,0 +1,449 @@
+  styleUrl: './posts.component.css'
+})
+export class PostsComponent implements OnInit{
+  posts: Post[];
+
+  ngOnInit(): void {
+    this.getAllPosts();
+  }
+
+  async getAllPosts(){
+    //@ts-ignore
+    this.posts = await directus
+      .request<Post[]>(readItems("posts", {
+        fields: ["id","title", "published_date", {author: ["name"]}]
+      }))
+  }
+}
+
+```
+When the component is initialized, it will retrieve all the posts using the `readItems()` method and store them in the `posts` array.
+
+To list the posts, replace the code in the `src/app/component/posts/posts.component.html` file with the following code:
+
+```html
+<h1>Blog Posts</h1>
+<ol>
+  <li *ngFor="let post of posts">
+    <a routerLink="#">
+      <h2>{{post.title}}</h2>
+    </a>
+    <span>
+      {{post.published_date}} &bull; {{post.author.name}}
+    </span>
+  </li>
+</ol>
+```
+### Add Routing for Posts
+Go to `src/app/app.routes.ts` file and add the following route in the `Routes` array:
+
+```ts
+ {path: 'blog', component: PostsComponent},
+```
+Once the application reloads, go to `http://localhost:4200/blog` and the list of posts will be displayed on the page.
+
+:::info Navigation
+
+In Angular, the order in which you put the routes in the `Routes` array will affect how components are loaded in your application. In this case, you don't want the path to `blog` to be consumed as a `slug`. As a result, ensure the blog route is put before slug in the Routes array.
+
+:::
+
+![blog post listing](post-listing.png)
+
+## Create Blog Post Pages
+You have learned how to create dynamic pages in a previous section, you will leverage the skill in this section to display individual post pages for the blog post listing.
+
+### Create a Component for Gallery Detail
+Navigate to your project directory in a terminal and create the post component:
+
+```bash
+ng g c component/post
+```
+Replace the code in the `src/app/component/post/post.component.ts` file with the following code.
+
+```ts
+import {Component, OnInit} from '@angular/core';
+import {directus, Post} from "../../../../directus";
+import {ActivatedRoute} from "@angular/router";
+import {CommonModule} from "@angular/common";
+import {readItem} from "@directus/sdk";
+
+@Component({
+  selector: 'app-post',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './post.component.html',
+  styleUrl: './post.component.css'
+})
+export class PostComponent implements OnInit{
+  post: Post;
+  baseUrl = "YOUR_DIRECTUS_URL";
+  constructor(private route: ActivatedRoute) {
+  }
+  ngOnInit(): void {
+    this
+      .getPostById(+this
+      .route
+      .snapshot
+      .paramMap.get('id'))
+  }
+
+  async getPostById(id: number){
+    //@ts-ignore
+    this.post = await directus
+      .request<Post>(readItem("posts", id));
+  }
+
+}
+```
+When the component is initialized, it will retrieve the path variable using the `ActivatedRoute` and pass it to the `readItem()` method to get the post with that id.
```

Comment:
```suggestion
When the component is initialized, it will retrieve the path variable using the `ActivatedRoute` and pass it to the `readItem()` function to get the post with that id.
```